setup-fortran.sh

```diff
@@ -504,3 +515,89 @@ install_intel()
       ;;
   esac
 }
+
+export_nvidiahpc_vars()
+{
+    local version=$1
+
+    # to convert version format from X.Y to X-Y
+    local cversion=$(echo "$version" | tr '.' '-')
+
+  cat >> $GITHUB_ENV <<EOF
+NVARCH=`uname -s`_`uname -m`;
+NVCOMPILERS=/opt/nvidia/hpc_sdk;
+MANPATH=$MANPATH:$NVCOMPILERS/$NVARCH/$cversion/compilers/man;
+PATH=$NVCOMPILERS/$NVARCH/$cversion/compilers/bin:$PATH;
+PATH=$NVCOMPILERS/$NVARCH/$cversion/comm_libs/mpi/bin:$PATH
+MANPATH=$MANPATH:$NVCOMPILERS/$NVARCH/$cversion/comm_libs/mpi/man
+EOF
+  for path in ${PATH//:/ }; do
+    echo $path >> $GITHUB_PATH
+  done
+}
+
+install_nvidiahpc_apt()
+{
+  local version=$1
+
+  # install environment-modules
+  install_environment_modules_apt
+
+  # to convert version format from X.Y to X-Y
+  local cversion=$(echo "$version" | tr '.' '-')
+
+  # install NVIDIA HPC SDK
+  echo "Installing NVIDIA HPC SDK $version..."
+  curl https://developer.download.nvidia.com/hpc-sdk/ubuntu/DEB-GPG-KEY-NVIDIA-HPC-SDK | sudo gpg --dearmor -o /usr/share/keyrings/nvidia-hpcsdk-archive-keyring.gpg
+  echo 'deb [signed-by=/usr/share/keyrings/nvidia-hpcsdk-archive-keyring.gpg] https://developer.download.nvidia.com/hpc-sdk/ubuntu/amd64 /' | sudo tee /etc/apt/sources.list.d/nvhpc.list
+  sudo apt-get update -y
+  sudo apt-get install -y nvhpc-$cversion
+  echo "NVIDIA HPC SDK $version installed."
+
+  # load NVIDIA HPC SDK module
+  echo "Loading NVIDIA HPC SDK $version module..."
+  NVCOMPILERS=/opt/nvidia/hpc_sdk; export NVCOMPILERS
+  export MODULEPATH=$NVCOMPILERS/modulefiles:$MODULEPATH
+  module load nvhpc
+  echo "NVIDIA HPC SDK $version module loaded."
+
+  # set environment variables
+  echo "Setting environment variables..."
+  export_nvidiahpc_vars $version
+
+  # set environment variables
+  export FC="nvfortran"
+  export CC="nvc"
+  export CXX="nvc++"
+  echo "Environment variables set."
+}
+
+install_nvidiahpc()
+{
+  local platform=$1
+  case $platform in
+    linux*)
+      install_nvidiahpc_apt $version
```

Comment:
Version 20.7 is the oldest version that can be found at https://developer.nvidia.com/nvidia-hpc-sdk-releases and is available on APT.

Reply:
https://developer.download.nvidia.com/hpc-sdk/ubuntu/amd64/

Reply:
Is there any limitation related to the specific versions you've chosen to add to the full test matrix? If not, I see no reason not to test/support them all, from 20.7 up to current

Reply:
It appears that versions 23.1, 22.7, 22.2, and possibly 21.1 encounter issues due to large data size during installation on the server.

Versions 20.7 and 20.9 are facing difficulties in locating the packages, requiring debugging to find the problem.

All other versions have been successfully installed.

The following are the outcomes of my tests on a private test repository:

**23.1**: 

```
dpkg: error processing archive /var/cache/apt/archives/nvhpc-23-1_23.1_amd64.deb (--unpack):
 cannot copy extracted data for './opt/nvidia/hpc_sdk/Linux_x86_64/23.1/profilers/Nsight_Systems/host-linux-x64/Mesa/libGL.so.1.5.0' to '/opt/nvidia/hpc_sdk/Linux_x86_64/23.1/profilers/Nsight_Systems/host-linux-x64/Mesa/libGL.so.1.5.0.dpkg-new': failed to write (No space left on device)
No apport report written because the error message indicates a disk full error
```

**22.7**:
```
dpkg: error processing archive /var/cache/apt/archives/nvhpc-22-7_22.7_amd64.deb (--unpack):
 cannot copy extracted data for './opt/nvidia/hpc_sdk/Linux_x86_64/22.7/profilers/Nsight_Compute/host/linux-desktop-glibc_2_11_3-x64/Plugins/libLinuxPPCPlatformPlugin.so' to '/opt/nvidia/hpc_sdk/Linux_x86_64/22.7/profilers/Nsight_Compute/host/linux-desktop-glibc_2_11_3-x64/Plugins/libLinuxPPCPlatformPlugin.so.dpkg-new': failed to write (No space left on device)
No apport report written because the error message indicates a disk full error
```

**22.2**:
```
cannot copy extracted data for './opt/nvidia/hpc_sdk/Linux_x86_64/22.2/profilers/Nsight_Systems/pkg/target-linux-x64/libcupti.so.11.2' to '/opt/nvidia/hpc_sdk/Linux_x86_64/22.2/profilers/Nsight_Systems/pkg/target-linux-x64/libcupti.so.11.2.dpkg-new': failed to write (No space left on device)
```

**21.1**:
```
File has unexpected size (2115251386 != 2115276192). Mirror sync in progress?
```

**20.9**:
```
E: Unable to locate package nvhpc-20-9
```

**20.7**:
```
E: Unable to locate package nvhpc-20-7
```

Reply:
Sounds good. We can still test them all in the full matrix, the failing ones will be reported as such when the bot updates the readme.